### PR TITLE
linter: added a check for casting to the real type and using the is_real function

### PR DESCRIPTION
**Pull request by i582**
See [deprecated core real](https://www.php.net/manual/ru/migration74.deprecated.php#migration74.deprecated.core.real).

### CODE REVIEW

File: src/linter/report.go [hunk 1]
```diff
@@ -628,6 +628,17 @@ echo $someVal;`,
 	20,
 ]`,
 		},
+
+		{
+			Name:     "langDeprecated",
+			Default:  false,
+			Quickfix: true,
+			Comment:  `Report the use of deprecated (per language spec) symbols.`,
```
Comment (quasilyte) [inline src/linter/report.go:636]: s/symbols/features/

File: src/tests/checkers/basic_test.go [hunk 2]
```diff
@@ -2126,3 +2126,22 @@ function f() {
...
+function f() {
+    $a = (real)100;
+    if (is_real($a)) {
+        echo 1;
+    }
+}
+`)
+	test.Expect = []string{
+		`do not cast to type real, use cast to float`,
+		`don't use is_real function, use is_float`,
```
Comment (quasilyte) [inline src/tests/checkers/basic_test.go:2144]: I would stick to this style of suggestion:
```
use is_float function instead of is_real
```

Comment (i582) [reply src/tests/checkers/basic_test.go]: Done.

File: src/linter/report.go (hunk 1 above)
Comment (i582) [reply src/linter/report.go]: Done.

**Review by quasilyte — APPROVED** [review]

**Pull request merged**